### ISSUE TITLE
WooCommerce plugin Sensei support?

### ISSUE THREAD

**Issue by Heichichi** [roots/sage-woocommerce#15]
## Submit a feature request or bug report

- [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/guidelines/blob/master/CONTRIBUTING.md)
- [x] This is a feature request
- [ ] This is a bug report
- [x] This request isn't a duplicate of an [existing issue](https://github.com/roots/sage-woocommerce/issues)
- [x] I've read the [docs](https://roots.io/sage/docs) and [NPM Debugging Guidelines post](https://discourse.roots.io/t/npm-debugging-guidelines-failed-npm-install-bower-install-or-gulp-build-read-this/3060) and followed them (if applicable)
- [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/c/sage) forums

**Please provide use cases for changing the current behavior:**

I installed your mod which works great! And Im wondering if its possible or maybe if I could modify your mod to be able to do the same thing with Sensei, which is a plugin from WooCommerce that is overriden in the same way as WooCommerce. Maybe you have some pointers on how to approach this?

**Comment by mmirus** [comment]
Hey @Heichichi,

Glad that Sage WooCommerce is working well for you. It's open source software, so you are welcome to use it as a starting point for a Sensei integration. It looks like Sensei has decent docs to help you get started. For example:

https://docs.woocommerce.com/document/sensei-theming/
https://docs.woocommerce.com/document/sensei-and-theme-compatibility/

I would start by creating a Blade file for one of Sensei's templates in your theme template folder in a 'sensei' subfolder and seeing if it loads that. Likely, you will need to filter `sensei_locate_template` to get it to find your Blade templates.

Once you have it detecting your loading your templates, it's a process of adapting the code from the plugin's default templates to Blade and troubleshooting any problems that may pop up along the way. Looking at the code in Sage WooCommerce may help you in that process.

If you [search the Roots Discourse](https://discourse.roots.io/search?q=plugin%20template) you can find threads giving pointers on how to override plugin templates that may also be helpful.

Since Sensei is a paid plugin, unfortunately I can't provide more specific advice or test anything for you.

**Comment by Heichichi** [comment]
Hi @mmirus,

I will start right away! Since its the same as woo Im hoping it wont be too much to change.. Could you explain very brief perhaps what happens when I installed your mod trough composer? I cant seem to find the file or where the code is inserted.

**Comment by mmirus** [comment]
I wouldn't advise making a plugin for this--I would put your code directly into Sage if you only need this for one or two sites and aren't planning to distribute it, or I would make a composer package like Sage WooCommerce that you can install into Sage if you want to be easier to reuse or distribute.

When you install Sage WooCommerce with `composer require roots/sage-woocommerce`, it just downloads this repository into your theme and tells Sage to load it. At the end of the day, the only change is that your theme loads https://github.com/roots/sage-woocommerce/blob/master/src/woocommerce.php.

Basically, `woocommerce.php` includes a bunch of filters that help WooCommerce work correctly with Sage, such as letting it know to use your Blade templates.

**Comment by Heichichi** [comment]
So Ive looked into the templating functions in the sensei plugin and adapted the ones I think is relevant in my mod. https://pastebin.com/6xaqtway If you can, maybe take a look and see if you think I have missed something? It was a while since I coded php and wordpress so just wanna make sure I dont do something completely wrong..

**Comment by Heichichi** [comment]
Template functions if you want to sneak a peak https://pastebin.com/zXHhHPr0

**Comment by mmirus** [comment]
If you followed the basic method used in Sage WooCommerce and it's working, it should be okay. Nothing jumped out at me from a very, very quick skim of your code.
